feat(iot-device): Add convenience method for acknowledging writable property update requests (#233)
Azure-Samples/azure-iot-samples-csharp: iot-hub/Samples/device/ConventionBasedOperations/readme.md

| [Telemetry](https://docs.microsoft.com/en-us/azure/iot-develop/concepts-convention#telemetry)             | :heavy_check_mark:  | :heavy_check_mark:  | :heavy_check_mark:  | :heavy_check_mark:  | :heavy_check_mark:  | The telemetry feature is an encapsulation over the existing device-to-cloud message send feature that the clients currently support.         |
| [Commands](https://docs.microsoft.com/en-us/azure/iot-develop/concepts-convention#commands)               | :heavy_check_mark:  | :heavy_check_mark:  | :heavy_check_mark:  | :heavy_check_mark:  | :heavy_minus_sign:  | The commands feature is an encapsulation over the existing direct methods feature that the clients currently support.                       |
| [Properties](https://docs.microsoft.com/en-us/azure/iot-develop/concepts-convention#writable-properties)  | :heavy_check_mark:  | :heavy_check_mark:  | :heavy_check_mark:  | :heavy_check_mark:  | :heavy_minus_sign:  | The properties feature is an encapsulation over the existing twin feature that the clients currently support.                                 |

> > Please try it out and let us know on our [discussions](https://github.com/Azure/azure-iot-sdk-csharp/discussions) page if you have any concerns, questions, or further changes you'd like to see!

Reviewer: ```suggestion
> Please try it out and let us know on our [discussions](https://github.com/Azure/azure-iot-sdk-csharp/discussions) page if you have any concerns, questions, or additional changes you would like to see!
```

> 

Reviewer: Not in your changes, but you may want to also update line 22 `master` -> `main`

Reviewer: same elsewhere

Author: good point, sure

### Commands

Commands subscription is implemented over the default handler. This gives rise to the following scenarios:
1. You've set both `SubscribeToCommandsAsync` and `SetMethodDefaultHandlerAsync`.
    
    Since both of these APIs set the default handler, the handler that is set the last will overwrite any previously set handler and that is the one which would always get triggered.

2. You've set `SubscribeToCommandsAsync` and a method-specific `SetMethodHandlerAsync`. 
    
    This will set the callback supplied to `SubscribeToCommandsAsync` to the default handler while the callback supplied to `SetMethodDefaultHandlerAsync` would be set for the method-specific handler. The order in which these callbacks are set doesn't matter in this case.
    - If you invoke the method (or command, in convention-aware terms) that was set via `SetMethodDefaultHandlerAsync`, `SetMethodDefaultHandlerAsync` callback would be triggered.
    - Any other method (or command, in convention-aware terms) would invoke the `SubscribeToCommandsAsync` callback.

### Properties

> Both convention-aware property updates (`SubscribeToWritablePropertyUpdateRequestsAsync`) and non-conventiom-aware twin desired property updates (`SetDesiredPropertyUpdateCallbackAsync`) are implemented over the default handler, i.e. property updates are not received on a per-property specific handler.

Reviewer: ```suggestion
Both convention-aware property updates (`SubscribeToWritablePropertyUpdateRequestsAsync`) and non-convention-aware twin desired property updates (`SetDesiredPropertyUpdateCallbackAsync`) are implemented over the default handler, i.e. property updates are not received on a per-property specific handler.
```

> Since both of these APIs set the default handler, the order in which these callbacks are set is relevant. The handler that is set the last will overwrite any previously set handler and that is the one which would always get triggered.

Reviewer: ```suggestion
Since both of these APIs set the default handler, the order in which these callbacks are set is relevant. The handler that is set the last will override any previously set handler and that is the one which would always get triggered.
```

Author: Since the callback handler is essentially getting replaced or overwritten, I feel like "overwrite" conveys the desired message. What do you say?

Reviewer: I don't feel strongly about it, just a suggestion, my reasoning is that if it's a text or a value it will be overwritten, if it's a method implementation, it's overridden :D

Reviewer: I discussed this with drwill and he agrees with you. Resolving.
